src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -519,9 +512,17 @@ public void PutAttachmentStream(DocumentsOperationContext context, Slice key, Sl
             _documentDatabase.Metrics.Attachments.BytesPutsPerSec.MarkSingleThreaded(stream.Length);
         }
 
-        private void DeleteAttachmentStream(DocumentsOperationContext context, Slice hash, int expectedCount = 1)
+        internal void DeleteAttachmentStream(DocumentsOperationContext context, Slice hash, int expectedCount = 1, bool fromReplication = false)
         {
-            if (GetCountOfAttachmentsForHash(context, hash) == expectedCount)
+            if (fromReplication)
+            {
+                AttachmentStreamsToDeleteAtTheEndOfBatch.Add(hash.Clone(context.Allocator));
+                return;
+            }
+
```

Comment:
Instead of the `fromReplication` maybe we can create some dummy reference to the stream?